### ISSUE TITLE
Bug: not all metadata values are recorded

### ISSUE THREAD

**Issue by martinbean** [muxinc/elements#500]
### Checklist

- [X] Verified whether your issue exists in the latest version Mux Element(s) release: `npm install @mux/[element_name]` or `yarn @mux/[element_name]`.
  - (NOTE: If the issue is not occurring in the latest version, we may not resolve it.)
- [X] Confirmed that this issue hasn't [already been filed](https://github.com/muxinc/elements/issues?q=is%3Aopen+is%3Aissue+label%3Abug).

### Which Mux Elements/Packages has a bug?

- common

### Which browser(s) are you using?

- Chrome
- Safari

### Which operating system(s) are you using?

macOS

### How are you using Mux Elements?

npm package

### Describe the bug

Not all metadata values are recorded with views. Particularly:

* `custom-1` and `custom-2`
* `sub-property-id`

## Steps To Reproduce

1. Add a `<mux-player>` element to your page.
2. Try setting the aforementioned properties, both with and without a `metadata-` prefix.

Link to code example:

https://codesandbox.io/s/musing-roentgen-885hzg?file=/index.html

## The current behavior

Values for custom dimensions and sub property ID are not recorded with views.

## The expected behavior

Values for custom dimensions and sub property ID to be recorded with views.

**Comment by dylanjha** [comment]
Hi @martinbean thanks for opening this. I can see why this is confusing.

All metadata fields can be set by the `metadata=` property on the element instance

```
const muxPlayer = document.querySelector('mux-player')
muxPlayer.metadata = {
  video_id: 'xxx',
  video_title: 'xxx',
  viewer_user_id: 'xxx',
  sub_property_id: 'xxx'
}
```

For the web component, there are 3 special Mux Data metadata attributes that can be used for convenience.

- `metadata-video-id`
- `metadata-video-title`
- `metadata-viewer-user-id`

These are the only shorthand attributes, but I'll see if we can expand the attributes to work with all Mux Data metadata fields, I can see how you would expect anything with the `metadata-` prefix to work.

**Comment by dylanjha** [comment]
@martinbean this is fixed on the [latest canary 1.4.0-canary.0-8ee139d](https://www.npmjs.com/package/@mux/mux-player/v/1.4.0-canary.0-8ee139d)

Please try it out! You should be able to provide any Mux Data metadata fields with the `metadata-` attribute prefix now.

**Comment by dylanjha** [comment]
@martinbean this is resolved as of 1.4!
